activation/handler_v2.go

```diff
@@ -367,13 +369,28 @@ func (h *HandlerV2) validatePreviousAtx(id types.NodeID, post *wire.SubPostV2, p
 		}
 		return min(prev.NumUnits, post.NumUnits), nil
 	case *wire.ActivationTxV2:
-		// TODO: support previous merged-ATX
-
-		// previous is solo ATX
-		if prev.SmesherID == id {
-			return min(prev.NiPosts[0].Posts[0].NumUnits, post.NumUnits), nil
+		if prev.MarriageATX != nil {
+			// Previous is a merged ATX
+			// need to find out if the given ID was present in the previous ATX
+			_, idx, err := identities.MarriageInfo(h.cdb, id)
```

Comment:
> Do we need a check her that prev.MarriageATX matches the one returned from identities.MarriageInfo?

I think it's not required for ATX syntactical validity. Having a different marriage ATX would mean that the ID married twice and it is already malicious. Requiring it would probably mean that we also need to do the same check for checkpointed ATXs. Besides, I want to remove decoding the previous ATX entirely here (see #6074)